pkgs/data/fonts/nerd-fonts/default.nix

```diff
@@ -0,0 +1,56 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchzip,
+}:
+
+let
+  releaseVersion = lib.trivial.importJSON ./release-version.json;
+  makeNerdFont =
+    {
+      pname,
+      dirName,
+      version,
+      longDescription,
+      license,
+      url,
+      sha256,
+      ...
+    }:
+    stdenvNoCC.mkDerivation {
+      inherit pname version;
+
+      src = fetchzip {
+        inherit url sha256;
+        stripRoot = false;
+      };
+
+      installPhase = ''
+        runHook preInstall
+
+        dst_opentype=$out/share/fonts/opentype/${dirName}
+        dst_truetype=$out/share/fonts/truetype/${dirName}
+
+        find -name \*.otf -exec mkdir -p $dst_opentype \; -exec cp -p {} $dst_opentype \;
+        find -name \*.ttf -exec mkdir -p $dst_truetype \; -exec cp -p {} $dst_truetype \;
+
+        runHook postInstall
+      '';
+
+      passthru.updateScript = ./update.sh;
+
+      meta = {
+        inherit longDescription license;
+        homepage = "https://nerdfonts.com/";
+        changelog = "https://github.com/ryanoasis/nerd-fonts/blob/v${releaseVersion}/changelog.md";
+        platforms = lib.platforms.all;
+        maintainers = with lib.maintainers; [ rc-zb ];
+      };
+    };
+in
+
+builtins.listToAttrs (
+  map (font: lib.attrsets.nameValuePair font.attrName (makeNerdFont font)) (
+    lib.trivial.importJSON ./fonts.json
+  )
+)
```

Comment:
```suggestion
symlinkJoin {
  name = "nerd-fonts";
  paths = map (font: makeNerdFont font) (lib.trivial.importJSON ./fonts.json);
}
```

Reply:
We need to produce a single derivation here, containing all fonts, right?

Reply:
I don't think bundling different fonts together (as in the legacy `nerdfonts`) is a good idea. They have different origins, different descriptions, different licenses and different tarballs. I.e. they have nothing to do with one another except that they live and get patched in the same [repo](https://github.com/ryanoasis/nerd-fonts/). Should we agglomerate them together, we would have to bundle many of the `kdePackages` together too, which is so meaningless as well as impractical.

Reply:
I meant that we remove the code for the old package altogether, we add the new packages, and we add a symlinkJoin with all of them as the replacement for the old package.

Reply:
No... Even if we manage to make such a mono-package, either it can't share the interface with its predecessor, or it shares the defeat as well -- difficulty to be built by Hydra due to its too many variants. On the other hand, the user may choose all the fonts by just appending the whole package set to their "fonts" or "packages" list.

Reply:
Well no, that's the thing of symlinkjoin, it just symlinks. So the package would be tiny, it just links to all the others.

Reply:
I don't know what Hydra actually do to build the cache. But if it was overwhelmed by the old package, which fetched only one GiB or two of tarballs, it must had made a huge number of random compositions on the overridable font choices. For the new package set, however, you can just mix the fonts in any manners as for other packages. Specifying font members and building a dedicated mono-derivation, lightweight or not, is quite redundant.